src/sv-pipeline/scripts/aggregate_picard.py

```diff
@@ -0,0 +1,276 @@
+def read_in(file_name, label):
+    started = False
+    newlist = []
+    wgsfile = file_name
+    
+    with tfio.gfile.GFile(file_name, "r") as inp:
+        for line in inp:
+            intable = line.startswith(label)
+            if intable == True:
+                started = True
+                continue
+            if started:
+                if line != '\n':
+                    newlist.append(line.rstrip('\n'))
+            if started and line == '\n':
+                return newlist
+                break
+                
+def newtable(file_name, label):
+    columnnames = read_in(file_name, label)[0].split("\t")
+    values = read_in(file_name, label)[1].split("\t")
+    dataframe = pd.DataFrame(columnnames).T
+    dataframe.columns = columnnames
+    dataframe_length = len(dataframe)
+    dataframe.loc[dataframe_length] = values
+    dataframe = dataframe.drop(0)
+    return dataframe
+
+def addrows(file_name, label):
+    onerow = newtable(file_name, label)
+    for i in range(len(read_in(file_name, label))):
+        nextrow = read_in(file_name, label)[i].split("\t")
```

Comment:
As above, here you are traversing the same file in each iteration - in general I/O is a bottleneck in many cases and we only want to read the files into memory once. 

You can rewrite this function to  read the file content into an array once and then fill in all the rows of the dataframe in one go.

Reply:
+1. Reading files from disk is orders of magnitude slower than in-memory operations. I'd put my money on this being where you're seeing the biggest slowdowns.